bodhi/server/services/updates.py

```diff
@@ -541,3 +549,36 @@ def new_update(request):
     result['caveats'] = caveats
 
     return result
+
+
+@update_waive_test_results.post(validators=(validate_update_id,
+                                            validate_acls,
+                                            validate_request),
+                                permission='edit', renderer='json',
+                                error_handler=bodhi.server.services.errors.json_handler)
+def waive_test_results(request):
+    """
+    Waive test results on a given update when gating is on.
+
+    Args:
+        request (pyramid.request): The current request.
+    Returns:
+        dict: A dictionary mapping the key "update" to the update.
+    """
+    update = request.validated['update']
+    comment = request.validated.pop('comment', None)
+
+    if update.locked:
+        request.errors.add('body', 'request',
+                           "Can't waive test results of a locked update")
```

Comment:
Since ```update.waive_test_results()``` below will also check for this condition, why don't we drop the check here and just catch the ```LockedUpdateException``` exception below?